compiler/optimizer/PreExistence.cpp

```diff
@@ -84,3 +83,52 @@ void TR_PrexArgInfo::dumpTrace() {
       }
    traceMsg(comp,  "</argInfo>\n");
 }
+
+static TR_PrexArgument *stronger(TR_PrexArgument *left, TR_PrexArgument *right, TR::Compilation *comp)
+   {
+   if (TR_PrexArgument::knowledgeLevel(left) > TR_PrexArgument::knowledgeLevel(right))
+      return left;
+   else if (TR_PrexArgument::knowledgeLevel(right) > TR_PrexArgument::knowledgeLevel(left))
+      return right;
+   else if (left && right)
+      {
+      if (left->getClass() && right->getClass())
+         {
+         if (comp->fe()->isInstanceOf(left->getClass(), right->getClass(), true, true, false))
+            return left;
+         else if (comp->fe()->isInstanceOf(right->getClass(), left->getClass(), true, true, false))
+            return right;
+         }
+      else if (left->getClass())
+         return left;
+      else if (right->getClass())
+         return right;
+
+      return NULL;
+      }
+   else
+      return left ? left : right;  // Return non-null prex argument when possible
+   }
+
+TR_PrexArgInfo *
+TR_PrexArgInfo::enhance(TR_PrexArgInfo *dest, TR_PrexArgInfo *source, TR::Compilation *comp)
+   {
+   // If dest is NULL, we can't simply return source, as TR_PrexArgInfo is mutable, any
+   // future change to dest will change source. Thus return a copy of source
+   //
+   if (!dest && source)
+      return new (comp->trHeapMemory()) TR_PrexArgInfo(source, comp->trMemory());
+   else if (!source)
+      return dest;
+
+   int32_t numArgsToEnhance = std::min(dest->getNumArgs(), source->getNumArgs());
```

Comment:
Are these two values expected to be different in any case ? The usage of min suggests so, but I'm just trying to get a clear statement of when that case occurs.

Reply:
I can't think of any case where they can be different. This is code from OpenJ9 written many years ago, not sure if this was to handle a case back then, but the two numbers should match otherwise the user is merging the wrong prex arg info. I'll add an assert to make sure the two numbers are the same